jmxtrans-output/jmxtrans-output-elastic/src/main/java/com/googlecode/jmxtrans/model/output/elastic/ElasticWriter.java

```diff
@@ -0,0 +1,237 @@
+
+		super(typeNames, booleanAsNumber, debugEnabled, settings);
+
+		this.rootPrefix = resolveProps(
+				firstNonNull(
+						rootPrefix,
+						(String) getSettings().get("rootPrefix"),
+						DEFAULT_ROOT_PREFIX));		
+
+		this.jsonFactory = new JsonFactory();
+		this.connectionUrl = connectionUrl;
+		this.indexName = this.rootPrefix + "_jmx-entries";
+
+	}
+
+	@Override
+	protected void internalWrite(Server server, Query query, ImmutableList<Result> results) throws Exception {
+		List<String> typeNames = this.getTypeNames();
+
+		for (Result result : results) {
+			log.debug("Query result: [{}]", result);
+			Map<String, Object> resultValues = result.getValues();
+			for (Entry<String, Object> values : resultValues.entrySet()) {
+				Object value = values.getValue();
+				if (isNumeric(value)) {
+					String message = createJsonMessage(server, query, typeNames, result, values, value);
+					log.debug("Insert into Elastic: Index: [{}] Type: [{}] Message: [{}]", indexName, TYPE_NAME, message);
+					Index index = new Index.Builder(message).index(indexName).type(TYPE_NAME).build();
+					jestClient.execute(index);
+				} else {
+					log.warn("Unable to submit non-numeric value to Elastic: [{}] from result [{}]", value, result);
+				}
+			}
+		}
+	}
+
+	private String createJsonMessage(Server server, Query query, List<String> typeNames, Result result, Entry<String, Object> values, Object value) throws IOException {
+
+		String keyString = getKeyString(server, query, result, values, typeNames, this.rootPrefix);
+
+		String alias;
+		if (server.getAlias() != null) {
+			alias = server.getAlias();
+		} else {
+			alias = server.getHost() + "_" + server.getPort();
+			alias = StringUtils.cleanupStr(alias);
+		}
+
+		Closer closer = Closer.create();
+		try {
+			ByteArrayOutputStream out = closer.register(new ByteArrayOutputStream());
+			JsonGenerator generator = closer.register(jsonFactory.createGenerator(out, UTF8));
+			generator.writeStartObject();
+			generator.writeStringField("server", alias);
+			generator.writeStringField("metric", keyString);
+			generator.writeNumberField("value", Double.parseDouble(value.toString()));
+			generator.writeStringField("resultAlias", result.getKeyAlias());
+			generator.writeStringField("attributeName", result.getAttributeName());
+			generator.writeStringField("key", values.getKey());
+			generator.writeNumberField("timestamp", result.getEpoch());
+			generator.writeEndObject();
+			generator.close();
+			return out.toString("UTF-8");
+		} catch (Throwable t) {
+			throw closer.rethrow(t);
+		} finally {
+			closer.close();
+		}
+	}
+
+	@VisibleForTesting
+	void setJestClient(JestClient jestClient) {
+		this.jestClient = jestClient;
+	}
+
+	@Override
+	public void validateSetup(Server server, Query query) throws ValidationException {
+		// no validations
+	}
+
+	@Override
+	public void start() throws LifecycleException {
+		super.start();
+
+		if (jestClient == null) {
```

Comment:
Seems that the creation of the JestClient would be better placed in the constructor. That way you can ensure the this object is always in a valid state. I know, we have already too much creation logic in the constructors of OutputWriters, but a cleanup is on the way...